### PR TITLE
118296117 check user authorized for posts

### PR DESCRIPTION
**Pull request by rustemmukhamadiev**

### CODE REVIEW

File: app/controllers/user_posts_controller.rb [hunk 1]
```diff
@@ -1,10 +1,14 @@
 class UserPostsController < ApplicationController
-  expose(:post, attributes: :post_params)
+  before_action :authenticate_user!, expect: %i(index)
```
Comment (vladimirbazhanov) [inline app/controllers/user_posts_controller.rb]: expect - may be except? :)

File: app/controllers/user_posts_controller.rb [hunk 2]
```diff
@@ -29,6 +33,10 @@ def destroy
 
   private
 
+  def authorize_user!
+    authorize post, :owner?
```
Comment (vladimirbazhanov) [inline app/controllers/user_posts_controller.rb]: let merge it now, but in future we need to discuss about policies

**Comment by vladimirbazhanov** [comment]
👍

File: app/controllers/user_posts_controller.rb (hunk 1 above)
Comment (rustemmukhamadiev) [inline app/controllers/user_posts_controller.rb]: @vladimirbazhanov thx, fixed

**Pull request merged**